jdbc/core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/statement/ShardingSpherePreparedStatement.java

```diff
@@ -653,7 +658,17 @@ public int[] executeBatch() throws SQLException {
         try {
             // TODO add raw SQL executor
             initBatchPreparedStatementExecutor();
-            return batchPreparedStatementExecutor.executeBatch(executionContext.getSqlStatementContext());
+            int[] results = batchPreparedStatementExecutor.executeBatch(executionContext.getSqlStatementContext());
+            if (statementOption.isReturnGeneratedKeys() && batchPreparedStatementExecutor.getStatements().size() == 1 && generatedValues.isEmpty()) {
+                ResultSet resultSet = batchPreparedStatementExecutor.getStatements().get(0).getGeneratedKeys();
```

Comment:
Why only get the result generated by the first statement?

Reply:
I put a condition which requires statements size must equal 1. when using write db or single db, the statements size is 1. when using sharding mode. the statements size may greater than 1. In sharding mode, We almost use keygenerators .So I think just supporting single mode and rw mode is enough. I also test get all statement result in sharding mode,it can work too. but I hope reduce scope of influence.